src/egithub_webhook.erl

```diff
@@ -275,3 +279,52 @@ line_comment_exists(Comments, Path, Position, Body) ->
     [] -> not_exists;
     [_|_] -> exists
   end.
+
+-spec handle_messages_or_review(egithub:credentials(), egithub:repository(),
+                                integer(), review() | [message()]) ->
+  egithub:result().
+handle_messages_or_review(Cred, Repo, PR, Messages) ->
+  {ok, ReviewStyle} = application:get_env(egithub, review_style, pr_review),
+  case ReviewStyle of
+    individual_comments ->
+      {ok, LineComments} = egithub:pull_req_comments(Cred, Repo, PR),
+      {ok, IssueComments} = egithub:issue_comments(Cred, Repo, PR),
+      Comments = LineComments ++ IssueComments,
+      write_comments(Cred, Repo, PR, Comments, Messages);
+    pr_review ->
+      dismiss_old_pr_reviews(Cred, Repo, PR),
+      write_pr_review(Cred, Repo, PR, Messages)
+  end.
+
+-spec write_pr_review(egithub:credentials(), egithub:repository(), integer(),
+                      review())->
+  egithub:result().
+write_pr_review(Cred, Repo, PR, PrReview) ->
+  egithub:pr_review(Cred, Repo, PR, PrReview, #{post_method => queue}).
+
+-spec dismiss_old_pr_reviews(egithub:credentials(), egithub:repository(),
+                            integer()) ->
+  ok.
+dismiss_old_pr_reviews(Cred, Repo, PR) ->
+  {ok, ExistingReviews} = egithub:pr_reviews(Cred, Repo, PR),
+  Fun =
+    fun(#{<<"id">> := RId,
+          <<"user">> := #{<<"login">> := <<"elvisci">>},
+          <<"state">> := State}) ->
+        case State of
+          <<"DISMISSED">> ->
+            % Don't try to dismiss an already dismissed review ;)
+            ok;
+          _ ->
+            Body = #{message => <<"Old/outdated review">>},
+            {ok, _} = egithub:dismiss_pr_review(Cred, Repo, PR, RId, Body)
+        end;
+       (_) ->
+        ok
+    end,
+  _ = lists:foreach(Fun, ExistingReviews),
+  % Without this sleep call, the dismissal message will appear below the
```

Comment:
OK, but then the sleep call should be done _outside_ this function. This function only dismisses the PR review, it doesn't need to know that we're doing something else afterwards.